src/Dapr.Client/DaprSubscribeConfigurationSource.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Autogenerated = Dapr.Client.Autogen.Grpc.v1;
+
+namespace Dapr.Client
+{
+    internal class DaprSubscribeConfigurationSource : ConfigurationSource
+    {
+        private AsyncServerStreamingCall<Autogenerated.SubscribeConfigurationResponse> call;
+        private string id = string.Empty;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="sourceStreamRequest">The request that is used to start the configuration streaming with Dapr.</param>
+        /// <param name="internalClient">The internal client used to call Dapr and begin a gRPC stream.</param>
+        /// <param name="callOptions">The options for the call to Dapr.</param>
+        internal DaprSubscribeConfigurationSource(
+            Autogenerated.Dapr.DaprClient internalClient,
+            Autogenerated.SubscribeConfigurationRequest sourceStreamRequest,
+            CallOptions callOptions) : base()
+        {
+            this.call = internalClient.SubscribeConfigurationAlpha1(sourceStreamRequest, callOptions);
+        }
+
+        /// <inheritdoc/>
+        public override string Id => id;
+
+        public override IAsyncEnumerator<IEnumerable<ConfigurationItem>> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+        {
+            return new DaprSubscribeConfigurationEnumerator(call, streamId => setIdIfNullOrEmpty(streamId), cancellationToken);
+        }
+
+        private void setIdIfNullOrEmpty(string id)
+        {
+            if (string.IsNullOrEmpty(this.id))
+            {
+                this.id = id;
+            }
+        }
+    }
+
+    internal class DaprSubscribeConfigurationEnumerator : IAsyncEnumerator<IEnumerable<ConfigurationItem>>
```

Comment:
This seems like a really complex type to handle - `IAsyncEnumerator<IEnumerable<ConfigurationItem>>` means I need a nested-for-loop to read all of the keys.

Is this complexity based on the wire-protocol?

Reply:
BTW since you're implementing your own IEnumerator it would be possible for us to "unwrap" this into a `IAsyncEnumerator<ConfigurationItem>` - would that be better?

Reply:
It's based off the wire values right now. Each response can contain a list of items. I'll see about unwrapping them as I agree it is a bit unwieldy.

Reply:
Playing around with this more, I don't think we can fully unwrap it without some complex logic on our side. Each call of `MoveNext` on the underlying stream gives us a new list of items. 

So we could track which item we're on in the list and then only call `MoveNext` when we're out of items but that feels like we could miss updates and like an easy place to introduce an error.